impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -331,24 +335,35 @@ else if (searchScope == LdapSearchScope.SUBTREE) {
     }
 
     private LdapContext createDefaultLdapContext() {
-        return createLdapContext(
+        try {
+          return createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 ldapIdentityStoreDefinition.bindDn(),
                 ldapIdentityStoreDefinition.bindDnPassword());
-    }
-
-    private static LdapContext createLdapContext(String url, String bindDn, String bindCredential) {
-        try {
-            return new InitialLdapContext(getConnectionEnvironment(url, bindDn, bindCredential), null);
-        } catch (AuthenticationException e) {
-            return null;
-        } catch (NamingException e) {
-            throw new IllegalStateException(e);
+        } catch (CommunicationException | InterruptedNamingException ce) {
```

Comment:
How did you come up with the list of exceptions to catch here, and in `createCallerLdapContext()`? Is it based on what the `InitialLdapContext` javadoc says it can throw? Or ... ?

I think the code is OK to submit as is, just wondering how you decided what to catch.